src/Console/ConfigurationResolver.php

```diff
@@ -419,7 +417,7 @@ public function getProgressType(): string
 
                 $this->progress = $progressType;
             } else {
-                $this->progress = ProgressOutputType::NONE;
+                $this->progress = $this->resolveProgressOutput();
```

Comment:
not sure if this change is intended.

originally, Progress was used only for human-usage, thus not when tool was used in chain of other tools, like using report format for tools, eg junit.
With that, progress bar was only indicated for txt format and verbose.

now, it's kind of used... always?
I see it as 2 changes, one is new Progress formatter (with change of default formatter), this line also start adding formatter for places we were not using the formatter (aka always).
We can consider this change (as same or separate PR), but let's make sure it doesn't look ugly.

Then, it's very confusing for me that this command use progress=dots:
`./php-cs-fixer check src/Doctrine --format=txt -v --show-progress=dots`
while this, progress=bar, even if we specified dots:
`./php-cs-fixer check src/Doctrine --format=json -v --show-progress=dots `
(because on original logic, progress is not reported at all when using non-txt format)
Yet, the CLI silently ignores it and provide other formatter


with above, i suggest to make partial revert of this PR, have new progress type available, and if we want to have progress bar available regardless of --format value, do it as separated PR, covering the mentioned cases

Reply:
As we discussed outside of Github, I'll fix it so it behaves like it should - progress bar as default (even without `--verbose`, but overridden by `--show-progress` when provided.

Reply:
Thanks for noticing this, it's fixed now and for non-txt reporters progress output is always `none`, as before. I also added assert for that in respective tests.

Reply:
👍🏻

Reply:
----

for long term, would you suggest to have non-txt always `none` (like now), or change it to have progress indicator (on stderr) also for non-txt reporters (on stdout) ?

Reply:
I think I would keep `none` as default for non-text reporters, since these are rather used for automation and progress is not needed there probably. Also, in long-running CI jobs the progress bar is not displayed properly but it is [multi-line with increasing progress in each line](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/actions/runs/7352046020/job/20016245340?pr=7603#step:10:15), because runners have time limit for line output. We can add support for `--show-progress` there, if requested.